### PR TITLE
[CIT-338] Split certificate module to multiple files

### PR DESCRIPTION
**Pull request by makr11st**
## Proposed changes

This code splits certificate module into its own module directory with proposed structure.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: mapper/thin_edge_json/Cargo.toml [hunk 1]
```diff
@@ -13,4 +13,4 @@ thiserror = "1.0"
 
 [dev-dependencies]
 pretty_assertions = "0.6"
-proptest = "0.10"
\ No newline at end of file
+proptest = "0.10"
```
Comment (makr11st) [inline mapper/thin_edge_json/Cargo.toml:16]: cargo fmt

File: tedge/src/services/mod.rs [hunk 2]
```diff
@@ -195,9 +195,9 @@ impl SystemCtlCmd {
     }
 }
 
-impl Into<String> for SystemCtlCmd {
-    fn into(self) -> String {
-        self.as_str().into()
+impl From<SystemCtlCmd> for String {
+    fn from(val: SystemCtlCmd) -> Self {
+        val.as_str().into()
```
Comment (makr11st) [inline tedge/src/services/mod.rs:200]: clippy

File: tedge/src/services/mod.rs [hunk 3]
```diff
@@ -214,9 +214,9 @@ impl SystemCtlParam {
     }
 }
 
-impl Into<String> for SystemCtlParam {
-    fn into(self) -> String {
-        self.as_str().into()
+impl From<SystemCtlParam> for String {
+    fn from(val: SystemCtlParam) -> Self {
+        val.as_str().into()
```
Comment (makr11st) [inline tedge/src/services/mod.rs:219]: clippy

File: tedge/src/cli/mod.rs [hunk 4]
```diff
@@ -1,8 +1,9 @@
-use crate::certificate;
+// use crate::certificate2;
```
Comment (PradeepKiruvale) [inline tedge/src/cli/mod.rs]: remove this commented code

**Review by PradeepKiruvale — COMMENTED** [review]
LGTM,  Is not it better to move the tests also into another separate file. Like tests.rs or something like that?

File: tedge/src/cli/certificate/cli.rs [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    fn build_command(self, context: BuildContext) -> Result<Box<dyn Command>, ConfigError> {
+        let config = context.config_repository.load()?;
+
+        let cmd = match self {
+            TEdgeCertCli::Create { id } => {
+                let cmd = CreateCertCmd {
+                    id,
+                    cert_path: config.query(DeviceCertPathSetting)?,
+                    key_path: config.query(DeviceKeyPathSetting)?,
+                    config_repository: context.config_repository,
```
[the remark this answers is not in the document]
Comment (mneumann) [reply tedge/src/cli/certificate/cli.rs:38]: You might not need `config_repository` in all commands, as now Didier derives the `device.id` within the `TEdgeConfig`, so no need to serialize `device.id`.

File: tedge/src/cli/certificate/cli.rs [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    /// The command will upload root certificate to Cumulocity.
+    C8y {
+        #[structopt(long = "user")]
+        /// Provided username should be a Cumulocity user with tenant management permissions.
+        /// The password is requested on /dev/tty, unless the $C8YPASS env var is set to the user password.
+        username: String,
+    },
+}
+
+impl BuildCommand for UploadCertCli {
```
Comment (mneumann) [inline tedge/src/cli/certificate/cli.rs:79]: I'd rather move the `build_command` implementation into the `BuildCommand` for `TEdgeCertCli`. This avoids loading the config twice.

File: tedge/src/cli/certificate/remove.rs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+impl RemoveCertCmd {
+    fn remove_certificate(&self, user_manager: &UserManager) -> Result<(), CertError> {
+        let _user_guard = user_manager.become_user(BROKER_USER)?;
+        std::fs::remove_file(&self.cert_path).or_else(ok_if_not_found)?;
+        std::fs::remove_file(&self.key_path).or_else(ok_if_not_found)?;
+
+        Ok(())
+    }
+
+    fn update_tedge_config(&self) -> Result<(), ConfigError> {
```
Comment (mneumann) [inline tedge/src/cli/certificate/remove.rs:42]: this will go away after #190

**Review by mneumann — COMMENTED** [review]
Please wait until PR #190 is merged. And rebase. Hope you'll not have too much fallout.

**Review by didier-wenzek — APPROVED** [review]
Nice to see the final result of this command refactoring.

**Pull request merged**